Community/GenericPresence/automation/jsr223/python/community/genericpresence/presence.py

```diff
@@ -0,0 +1,159 @@
+        - events: Access to the events Object so we can call sendCommand.
+        - proxy_name: Name of the proxy Item that we want to set to away.
+        - flap_time: Number of minutes we wait before marking the person away.
+    """
+    log.info("{} has been away for {} minutes, setting to away."
+            .format(proxy_name,flap_time))
+    events.sendCommand(proxy_name, "OFF")
+
+@rule("Presence",
+      description=("Aggregates presence sensor readings and implements "
+                   "anti-flapping for away."),
+      tags=["presence"])
+@trigger_generator(presence_config)
+def presence(event):
+    """
+    Rule that implements presence detection based on the state of sensors. The
+    Rule assumes there is a proxy Item associated with each sensor which
+    represents the presence for that sensor. There is flapping logic which
+    requires the user to be away for some minutes before the proxy will be set
+    to OFF.
+    """
+
+    # Get the proxy Item associated with the sensor.
+    try:
+        sensor_name = event.itemName
+        proxy_name = presence_config[sensor_name][0]
+        flap_time = presence_config[sensor_name][1]
+
+    except ValueError:
+        presence.log.error("{} is not in presence_sensors!"
+                           .format(event.itemName))
+    except IndexError:
+        presence.log.error("presence_items does not have the same number of "
+                           "elements as presence_sensors, check your "
+                           "configuration!")
+
+    else:
+        # Item and sensor are the same, cancel the flapping timer if there is
+        # one.
+        if items[sensor_name] == items[proxy_name]:
+            if (proxy_name in presence_timers.keys()
+                    and not presence_timers[proxy_name].hasTerminated()):
+                presence.log.debug("{} came home, cancelling the flapping "
+                                   "timer.".format(proxy_name))
+                presence_timers[proxy_name].cancel()
+                del presence_timers[proxy_name]
+            return
+
+        # Item and sensor are different, set a flapping timer if the sensor went
+        # to OFF.
+        if event.itemState == OFF:
+            presence.log.debug("Setting the flapping timer for {}"
+                               .format(proxy_name))
+            presence_timers[proxy_name] = ScriptExecution.createTimer(
+                DateTime.now().plusMinutes(presence_flap_time),
+                lambda: away(presence.log, events, proxy_name, presence_timers))
+        elif event.itemState == ON:
+            presence.log.debug("{} came home.".format(proxy_name))
+            events.sendCommand(proxy_name, "ON")
+
+def scriptUnloaded():
+    """
+    Cancels all the timers when the script is unloaded to avoid timers hanging
+    around.
+    """
+    if not presence_timers:
+        for key in presence_timers.keys():
+            if not presence_timers[key].hasTerminated()
```

Comment:
```suggestion
            if not presence_timers[key].hasTerminated():
```

Reply:
This one too was fixed in the commit I forgot to push. I actually changed to use list comprehension instead of nesting the if inside the for loop.